pgjdbc/src/main/java/org/postgresql/util/StreamWrapper.java

```diff
@@ -67,16 +79,26 @@ public StreamWrapper(InputStream stream) throws PSQLException {
           throw e;
         }
         // The finalize action is not created if the above code throws
-        finalizeAction = new StreamWrapperFinalizeAction(tempFile);
         this.offset = 0;
         this.length = rawData.length + diskLength;
         this.rawData = null;
         this.stream = null; // The stream is opened on demand
+        finalizeAction = new StreamWrapperCleaningAction(tempFile);
+        @SuppressWarnings("argument")
+        LazyCleaner.Cleanable cleanable = LazyCleaner.getInstance().register(this, finalizeAction);
+        cleaner = cleanable;
+
       } else {
+        // create a null lambda to avoid checker errors
+        finalizeAction = (LazyCleaner.CleaningAction)(boolean clean) -> { };
         this.rawData = rawData;
         this.stream = null;
         this.offset = 0;
         this.length = rawData.length;
+        @SuppressWarnings("argument")
+        LazyCleaner.Cleanable cleanable = LazyCleaner.getInstance().register(this, finalizeAction);
```

Comment:
There's no need to create the cleaning action if there's nothing to clean

Reply:
Currently we don't check for a null cleaning action. We could and then get rid of this.

Reply:
I've removed these now